api/ssi-bridge/src/services/ssi-service.ts

```diff
@@ -84,164 +69,175 @@ export class SsiService {
 		try {
-			const key: Identity.KeyPair = Identity.KeyPair.fromJSON(identity.key);
-			const doc = Document.fromJSON(identity.doc) as any;
-
+			const decodedKey = {
+				public: Array.from(bs58.decode(identity.keys.sign.public)),
+				secret: Array.from(bs58.decode(identity.keys.sign.private))
+			};
+			const json = {
+				type: identity.keys.sign.type,
+				public: decodedKey.public,
+				private: decodedKey.secret
+			};
+			const key: Identity.KeyPair = KeyPair.fromJSON(json);
+			const { document, messageId } = await this.getLatestIdentityDoc(identity.id);
 			return {
-				doc,
-				key
+				document,
+				key,
+				messageId
 			};
 		} catch (error) {
 			this.logger.error(`Error from identity sdk: ${error}`);
 			throw new Error('could not parse key or doc of the identity');
 		}
 	}
 
-	generateIdentity() {
+	async generateIdentity(): Promise<{ account: Identity.Account; doc: Identity.Document; key: Identity.KeyPair }> {
 		try {
-			const { doc, key } = new Document(this.config.keyType, Identity.Network.mainnet().toString()) as any;
+			const builder = this.getAccountBuilder();
+			const keyPair = new KeyPair(KeyType.Ed25519);
+			const account = await builder.createIdentity({ privateKey: keyPair.private() });
 
 			return {
-				doc,
-				key
+				account,
+				doc: account.document(),
+				key: keyPair
 			};
 		} catch (error) {
 			this.logger.error(`Error from identity sdk: ${error}`);
-			throw new Error(`could not create identity document from keytype: ${this.config.keyType}`);
+			throw new Error(`could not create identity document from keytype: ${KeyType.Ed25519}`);
 		}
 	}
-	getKeyCollectionTag = (keyCollectionIndex: number) => `${this.config.keyCollectionTag}-${keyCollectionIndex}`;
+	getBitmapTag = (id: string, bitmapIndex: number) => `${id}#${this.config.bitmapTag}-${bitmapIndex}`;
+
+	private getAccountBuilder(usePermaNode?: boolean): Identity.AccountBuilder {
+		const clientConfig: Identity.IClientConfig = this.getConfig(usePermaNode);
+		const builderOptions = {
+			clientConfig
+		};
+		return new AccountBuilder(builderOptions);
+	}
 
 	private getIdentityClient(usePermaNode?: boolean) {
-		const cfg = Identity.Config.fromNetwork(Identity.Network.mainnet());
-		if (usePermaNode) {
-			cfg.setPermanode(this.config.permaNode);
-		}
-		cfg.setNode(this.config.node);
+		const cfg: Identity.IClientConfig = this.getConfig(usePermaNode);
 		return Client.fromConfig(cfg);
 	}
 
-	private addPropertyToDoc(doc: Identity.Document, property: { [key: string]: any }): Identity.Document {
-		return Identity.Document.fromJSON({
-			...doc.toJSON(),
-			...property
-		});
+	private getConfig(usePermaNode?: boolean): Identity.IClientConfig {
+		return {
+			permanodes: usePermaNode ? [{ url: this.config.permaNode }] : [],
+			primaryNode: { url: this.config.node },
+			network: Identity.Network.mainnet(),
```

Comment:
I will take a note how to handle this properly... testnet can be used by adjusting the nodes i think this will mainly affect if the did starts with did:iota:.. or did:iota-test:...